posthog/api/test/test_survey.py

```diff
@@ -92,9 +92,11 @@ def test_can_create_survey_with_linked_flag_and_targeting(self):
         )
 
         response_data = response.json()
+
         assert response.status_code == status.HTTP_201_CREATED, response_data
         assert response_data["linked_flag"]["id"] == notebooks_flag.id
         assert FeatureFlag.objects.filter(id=response_data["targeting_flag"]["id"]).exists()
+        self.assertNotEqual(response_data["targeting_flag"]["key"], "survey-targeting-power-users-survey")
```

Comment:
can you also test the matching case? Perhaps a regex match for the id, to ensure this is generated correctly?

Reply:
done!